### ISSUE TITLE
configuration change is not detected when it is coming from `extends`

### ISSUE THREAD

**Issue by Sysix** [oxc-project/oxc#10373]
The current behavior of the language server is to revalidate the diagnostics when some `.oxlintrc.json` file is changing.
This works perfectly with nested configuration.

We are sending a `workspace/didChangeWatchedFiles` request from VSCode to the server:
https://github.com/oxc-project/oxc/blob/fbd4f92f15404ef558b7d29172eeb33fd8d3fa0f/editors/vscode/client/extension.ts#L200

## Description

When some `.oxlintrc.json` has `extends` keyword, and the files included in them are changing, the language server will not revalidate the diagnostics.

## Setup

`/.oxlintrc.json`:

```
{
  "extends": ["./configs/.oxlintrc.json"]
}
```

`/configs/.oxlintrc.json`:

```
{
  "rules": { ... }
}
```

## Expected

Changing `./configs/.oxlintrc.json` should revalidate the diagnostics on the language server.
This will be a problem because the `extends` can include any filename the user wants, as long as it is valid.
